runtime/runtime.go

```diff
@@ -411,6 +411,14 @@ func (rt *Runtime) Serve(ctx context.Context) error {
 		return fmt.Errorf("at least one address must be configured in runtime parameters")
 	}
 
+	if rt.Params.Addrs != nil {
```

Comment:
Maybe we should warn if it's nil, if that indicates that no arguments were set...?

Reply:
I can do a bit more digging but afaik `rt.Params.Addrs` is actually never `nil` because it will default to `:8181` unless specified with an  `--addr` :disappointed:

Reply:
Does the flag package not have a way to discern unset flags from flags set to the default value?

Reply:
Interesting, did not consider this, I will check :+1:

Reply:
https://pkg.go.dev/flag#Visit I am going to look at something like this potentially as based on some searching last night seems like the most viable option :thinking: 

What do you guys think? :+1:

Reply:
Go for it!

Reply:
Hey :wave: 

Have not forgotten about this sorry, I have something semi-good looking with @srenatus solution, will do some more testing tomorrow and hopefully push something before FOSDEM!

Ty!

Reply:
Hey Team :wave:!

Just wondering if you guys could take a look at the latest change for this, I think I am on the right track but I can't seem to get it to work as I would like it to.

I'm going based on this: https://github.com/spf13/cobra/issues/453#issuecomment-303548563

But it seems to just always print it.. feel like I am missing something here :thinking: 

```shell
➜  opa git:(update-security-docs-bindings) opa run --server
{"level":"info","msg":"OPA running on a public (0.0.0.0) network interface. Unless you intend to expose OPA outside of the host, binding to the localhost interface (--addr localhost:8181) is recommended. See https://www.openpolicyagent.org/docs/latest/security/#interface-binding","time":"2023-02-06T11:45:01+01:00"}
{"addrs":[":8181"],"diagnostic-addrs":[],"level":"info","msg":"Initializing server.","time":"2023-02-06T11:45:01+01:00"}
^C{"level":"info","msg":"Shutting down...","time":"2023-02-06T11:45:02+01:00"}
{"level":"info","msg":"Server shutdown.","time":"2023-02-06T11:45:02+01:00"}
➜  opa git:(update-security-docs-bindings) opa run --server --addr 0.0.0.0:8181
{"level":"info","msg":"OPA running on a public (0.0.0.0) network interface. Unless you intend to expose OPA outside of the host, binding to the localhost interface (--addr localhost:8181) is recommended. See https://www.openpolicyagent.org/docs/latest/security/#interface-binding","time":"2023-02-06T11:45:10+01:00"}
{"addrs":["0.0.0.0:8181"],"diagnostic-addrs":[],"level":"info","msg":"Initializing server.","time":"2023-02-06T11:45:10+01:00"}
^C{"level":"info","msg":"Shutting down...","time":"2023-02-06T11:45:12+01:00"}
{"level":"info","msg":"Server shutdown.","time":"2023-02-06T11:45:12+01:00"}
```

Thanks in advance! :smile:

Reply:
Hey guys! :wave: 

Think I have solved this, looks like I needed to use

```golang
		PreRun: func(cmd *cobra.Command, args []string) {
			if cmd.Flags().Changed("addr") {
				cmdParams.rt.AddrChanged = true
			}
		},
```

Something like this, am doing some further testing and will report back :+1:

Reply:
Cool! I don't know exactly why that is, but it's cool that you've found a solution.

Nit: 
```go
cmdParams.rt.AddrChanged = cmd.Flags().Changed("addr")
```

Reply:
Ah @srenatus my mistake, I actually miss-understood the way PreRun and Run works, you can just handle this in the Run command itself, so it would just look like this:

```golang
		Run: func(cmd *cobra.Command, args []string) {
			cmdParams.rt.AddrChanged = cmd.Flags().Changed("addr")
			ctx := context.Background()
			rt, err := initRuntime(ctx, cmdParams, args)
			if err != nil {
				fmt.Println("error:", err)
				os.Exit(1)
			}
			startRuntime(ctx, rt, cmdParams.serverMode)
		}
```

Will tidy up and push later today :+1: 

TIL lots about Cobra and Flags... :sweat_smile: